src/app/view/perspectives/search/all/search-all.component.ts

```diff
@@ -48,42 +48,50 @@ export class SearchAllComponent implements OnInit, OnDestroy {
 
   public views: ViewModel[];
 
-  private querySubscription: Subscription;
-
-  private collectionsSubscription: Subscription;
-
-  private documentsSubscription: Subscription;
-
-  private viewsSubscription: Subscription;
+  private subscriptions = new Subscription();
 
   constructor(private searchService: SearchService,
               private store: Store<AppState>) {
   }
 
   public ngOnInit() {
-    this.querySubscription = this.store.select(selectNavigation)
-      .pipe(
-        filter((navigation: NavigationState) => {
-          return Boolean(navigation && navigation.workspace && navigation.workspace.organizationCode && navigation.workspace.projectCode)
-        }),
-        map(navigation => navigation.query),
-        filter(query => !isNullOrUndefined(query)),
-        map(query => ({...query, page: 0, pageSize: 100})), // TODO implement pagination logic
-        tap(query => this.store.dispatch(new CollectionsAction.Get({query}))),
-        tap(query => this.store.dispatch(new DocumentsAction.Get({query}))),
-        tap(query => this.store.dispatch(new ViewsAction.Get({query})))
-      ).subscribe();
-
-    this.collectionsSubscription = this.store.select(selectCollectionsByQuery).subscribe(collections => {
-      this.collections = collections;
+    [
+      this.subscribeOnQuery(),
+      this.subscribeOnCollection(),
+      this.subscribeOnDocuments(),
+      this.subscribeOnViews()
+    ].forEach(subscription => this.subscriptions.add(subscription));
+  }
+
+  private subscribeOnQuery() {
+    return this.store.select(selectNavigation).pipe(
+      filter((navigation: NavigationState) => {
+        return Boolean(navigation && navigation.workspace && navigation.workspace.organizationCode && navigation.workspace.projectCode);
+      }),
+      map(navigation => navigation.query),
+      filter(query => !isNullOrUndefined(query)),
+      map(query => ({...query, page: 0, pageSize: 100})), // TODO implement pagination logic
+      tap(query => this.store.dispatch(new CollectionsAction.Get({query}))),
```

Comment:
Collections are always queried before the page is loaded. So we don't need to load them again.

Reply:
And it is the same for views after the latest changes.